testproject/Assets/Tests/Runtime/MultiprocessRuntime/ExecuteStepInContext.cs

```diff
@@ -0,0 +1,289 @@
+    {
+        m_StartTime = Time.time;
+        m_IsRegistering = isRegistering;
+        m_ActionContext = actionContext;
+        m_StepToExecute = stepToExecute;
+        m_WaitMultipleUpdates = waitMultipleUpdates;
+        m_IgnoreTimeoutException = ignoreTimeoutException;
+
+        if (additionalIsFinishedWaiter != null)
+        {
+            m_AdditionalIsFinishedWaiter = additionalIsFinishedWaiter;
+        }
+
+        if (networkManager == null)
+        {
+            networkManager = NetworkManager.Singleton;
+        }
+
+        m_NetworkManager = networkManager; // todo test using this for multiinstance tests too?
+
+        var callerMethod = new StackFrame(1).GetMethod(); // one skip frame for current method
+
+        var methodId = GetMethodIdentifier(callerMethod); // assumes called from IEnumerator MoveNext, which should be the case since we're a CustomYieldInstruction. This will return a generated class name which should be unique
+        if (!s_MethodIDCounter.ContainsKey(methodId))
+        {
+            s_MethodIDCounter[methodId] = 0;
+        }
+
+        string currentActionID = $"{methodId}-{s_MethodIDCounter[methodId]++}";
+        m_CurrentActionID = currentActionID;
+
+        if (m_IsRegistering)
+        {
+            Assert.That(allActions, Does.Not.Contain(currentActionID)); // sanity check
+            allActions[currentActionID] = this;
+        }
+        else
+        {
+            if (ShouldExecuteLocally)
+            {
+                m_StepToExecute.Invoke(paramToPass);
+            }
+            else
+            {
+                if (networkManager.IsServer)
+                {
+                    TestCoordinator.Instance.TriggerActionIDClientRpc(currentActionID, paramToPass,
+                        clientRpcParams: new ClientRpcParams
+                        {
+                            Send = new ClientRpcSendParams { TargetClientIds = TestCoordinator.AllClientIdsExceptMine.ToArray() }
+                        });
+                    foreach (var clientId in TestCoordinator.AllClientIdsExceptMine)
+                    {
+                        m_ClientIsFinishedChecks.Add(TestCoordinator.ConsumeClientIsFinished(clientId));
+                    }
+                }
+                else
+                {
+                    throw new NotImplementedException();
+                }
+            }
+        }
+    }
+
+    public void Invoke(byte[] args)
+    {
+        m_StepToExecute.Invoke(args);
+        if (!m_WaitMultipleUpdates)
+        {
+            if (!m_NetworkManager.IsServer)
+            {
+                TestCoordinator.Instance.ClientFinishedServerRpc();
+            }
+            else
+            {
+                throw new NotImplementedException("todo implement");
+            }
+        }
+    }
+
+    public override bool keepWaiting
```

Comment:
again, we apparently didn't write rules for property names (yet) but my gut tells me that it's first letter should've been capitalized.
```suggestion
    public override bool KeepWaiting
```

Reply:
unfortunately, this is overriding a property we don't control :(